hoomd/hpmc/compute.py

```diff
@@ -329,19 +345,37 @@ def betaP(self):  # noqa: N802 - allow function name
             In MPI parallel execution, `betaP` is available on rank 0 only.
             `betaP` is `None` on ranks >= 1.
         """
+        compression_contribution = 0
+        expansion_contribution = 0
+        n_fit = int(numpy.ceil(self.xmax / self.dx))
+        box = self._simulation.state.box
+        N = self._simulation.state.N_particles
+        rho = N / box.volume
+        if numpy.isnan(self.sdf).all() and numpy.isnan(
+                self.sdf_expansion).all():
+            return None
         if not numpy.isnan(self.sdf).all():
             # get the values to fit
-            n_fit = int(numpy.ceil(self.xmax / self.dx))
             sdf_fit = self.sdf[0:n_fit]
             # construct the x coordinates
             x_fit = numpy.arange(0, self.xmax, self.dx)
             x_fit += self.dx / 2
             # perform the fit and extrapolation
             p = numpy.polyfit(x_fit, sdf_fit, 5)
-
-            box = self._simulation.state.box
-            N = self._simulation.state.N_particles
-            rho = N / box.volume
-            return rho * (1 + numpy.polyval(p, 0.0) / (2 * box.dimensions))
-        else:
-            return None
+            p0_compression = numpy.polyval(p, 0.0)
+            compression_contribution = rho * p0_compression / (2
+                                                               * box.dimensions)
+        if not numpy.isnan(self.sdf_expansion).all():
```

Comment:
Why are any of the sdf values `nan` in the first place? They are initialized to 0.

Reply:
That was already in the code, so I can't say for sure why it was added. But you're right, they're initialized to 0 and the only way I see they could become `nan` is if `m_pdata->getNGlobal() * m_dx == 0`.